webapp/src/components/Board/PieceComponent.jsx

```diff
@@ -0,0 +1,157 @@
+import React, { useContext } from 'react';
+import axios from 'axios';
+import BoardContext from './BoardContext';
+
+const PieceComponent = ({
+	skin,
+	color,
+	name,
+	selected,
+	targeted,
+	idle,
+	index,
+	turn,
+	access,
+	from,
+	lobbyID,
+	chess,
+}) => {
+	/**
+	 * Вернет true, если клетка черная
+	 * @param {int} ind 0 - 63
+	 * @returns
+	 */
+	const isBlackCell = (ind) => {
+		const row = Math.floor(ind / 8);
+		const col = ind % 8;
+		return (row % 2 === 0 && col % 2 === 1) || (row % 2 === 1 && col % 2 === 0);
+	};
+	/**
+	 * Преобразует индекс клетки в классическую нотацию
+	 * @param {int} ind 0 - 63
+	 * @returns
+	 */
+	const getCODE = (ind) => {
+		const codes = ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h'];
+		return codes[ind % 8] + String(8 - Math.floor(ind / 8));
+	};
+
+	/**
+	 * Преобразует индекс клетки из классической нотации в число от 0 до 63
+	 * @param {String} code
+	 * @returns
+	 */
+	const getIND = (code) => {
+		const pos = String(code);
+
+		const codes = {
+			h: 7,
+			g: 6,
+			f: 5,
+			e: 4,
+			d: 3,
+			c: 2,
+			b: 1,
+			a: 0,
+		};
+		return codes[pos[0]] + (8 - Number(pos[1])) * 8;
+	};
+
+	//   TODO: добавить плучение svg запросом
+	const src = {
```

Comment:
`src = { axios.get('svg') }`
would get called with every call of Component.render()